File: mog-rs/factory/sqlserver-tables-to-python-dataclass/tests/expected.py
from __future__ import annotations
from dataclasses import dataclass
from datetime import date, datetime, time
from decimal import Decimal
from typing import Any
from uuid import UUID

@dataclass
class CsgenOrders:
    order_id: int
    quantity: int
    small_count: int
    tiny_flag: int
    customer_code: str
    description: str | None
    legacy_note: str | None
    fixed_code: str | None
    region_code: str | None
    long_text: str | None
    is_active: bool
    discount_pct: Decimal
    unit_price: Decimal
    ratio: float
    small_ratio: float | None
    created_at: datetime
    updated_at: datetime | None
    legacy_ts: datetime | None
    order_date: date
    order_time: time | None
    order_uid: UUID
    payload: bytes | None
    extra_props: Any | None # TODO(mog): SQL Server sql_variant; runtime-typed (typing.Any), verify mapping.

File: mog-rs/factory/sqlserver-tables-to-csharp/tests/expected.cs
public class CsgenOrders
{
    public long OrderId { get; set; }
    public int Quantity { get; set; }
    public short SmallCount { get; set; }
    public byte TinyFlag { get; set; }
    public string CustomerCode { get; set; }
    public string? Description { get; set; }
    public string? LegacyNote { get; set; }
    public string? FixedCode { get; set; }
    public string? RegionCode { get; set; }
    public string? LongText { get; set; }
    public bool IsActive { get; set; }
    public decimal DiscountPct { get; set; }
    public decimal UnitPrice { get; set; }
    public double Ratio { get; set; }
    public float? SmallRatio { get; set; }
    public DateTime CreatedAt { get; set; }
    public DateTimeOffset? UpdatedAt { get; set; }
    public DateTime? LegacyTs { get; set; }
    public DateOnly OrderDate { get; set; }
    public TimeOnly? OrderTime { get; set; }
    public Guid OrderUid { get; set; }
    public byte[]? Payload { get; set; }
    public string? ExtraProps { get; set; } // TODO(mog): SQL Server variant type has no fixed C# type; stored as string.
}
